dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/imports/TrackerImportReportTest.java

```diff
@@ -0,0 +1,134 @@
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.tracker.imports;
+
+import static org.hisp.dhis.webapi.controller.tracker.JsonAssertions.assertReportEntities;
+
+import java.util.List;
+import java.util.Set;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.relationship.RelationshipType;
+import org.hisp.dhis.trackedentity.TrackedEntityType;
+import org.hisp.dhis.tracker.TrackerType;
+import org.hisp.dhis.tracker.imports.report.ImportReport;
+import org.hisp.dhis.web.HttpStatus;
+import org.hisp.dhis.web.WebClient;
+import org.hisp.dhis.webapi.DhisControllerConvenienceTest;
+import org.hisp.dhis.webapi.controller.tracker.JsonImportReport;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests {@link ImportReport} behavior through {@link TrackerImportController} using (mocked) REST
+ * requests
+ */
+class TrackerImportReportTest extends DhisControllerConvenienceTest {
+
+  private static final String ORG_UNIT_UID = "PSeMWi7rBgb";
+
+  private static final String TE_TYPE_UID = "PrZMWi7rBga";
+
+  private static final String PROGRAM_UID = "XrZRFi7rU9e";
+
+  private static final String PS_UID = "I8RRFi7rUps";
+
+  private static final String REL_TYPE_UID = "A4ORFi7rReL";
+
+  @BeforeEach
+  void setUp() {
+    TrackedEntityType type = createTrackedEntityType('A');
+    type.setUid(TE_TYPE_UID);
+    manager.save(type);
+
+    OrganisationUnit orgUnit = createOrganisationUnit('A');
+    orgUnit.setUid(ORG_UNIT_UID);
+    manager.save(orgUnit);
+
+    Program program = createProgram('A');
+    program.setOrganisationUnits(Set.of(orgUnit));
+    program.setTrackedEntityType(type);
+    program.setUid(PROGRAM_UID);
+    manager.save(program);
+
+    ProgramStage programStage = createProgramStage('A', program);
+    programStage.setUid(PS_UID);
+    manager.save(programStage);
+
+    RelationshipType relType = createPersonToPersonRelationshipType('A', program, type, false);
+    relType.setUid(REL_TYPE_UID);
+    manager.save(relType);
+  }
+
+  @Test
+  void shouldReturnOrderedEntitiesInReportWhenFlatPayloadImportIsSuccessful() {
+    JsonImportReport importReport =
+        POST(
+                "/tracker?async=false&reportMode=FULL"
+                    + "&validationMode=SKIP"
+                    + "&atomicMode=OBJECT"
+                    + "&skipRuleEngine=true",
+                WebClient.Body("tracker/import_flatten_payload.json"))
+            .content(HttpStatus.OK)
+            .as(JsonImportReport.class);
+
+    assertReportEntities(
+        List.of("IybbQIQt6te", "daMwzsKN3te", "FRM97UKN8te"),
+        importReport,
```

Comment:
nit: the uids you expect are tracked entity ones I assume right? So might be bit clearer to move the tracker type parameter second so its closer to the expected uids 😄

Reply:
:+1: